cli/src/program.rs

```diff
@@ -2627,6 +2632,70 @@ fn check_payer(
     Ok(())
 }
 
+// This enum is equivalent to an Option but was added to self-document
+// the ok variants and has the benefit of not forcing the caller to use
+// the result if they don't care about it.
+enum UpdateComputeUnitLimitResult {
+    UpdatedInstructionIndex(usize),
+    NoInstructionFound,
+}
+
+// Returns the index of the compute unit limit instruction
+fn simulate_and_update_compute_unit_limit(
+    rpc_client: &RpcClient,
+    transaction: &mut Transaction,
+) -> Result<UpdateComputeUnitLimitResult, Box<dyn std::error::Error>> {
+    let Some(compute_unit_limit_ix_index) = transaction
+        .message
+        .instructions
+        .iter()
+        .enumerate()
+        .filter_map(|(ix_index, instruction)| {
```

Comment:
nit: I believe this could be `find_map` instead https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.find_map

> iter.find_map(f) is equivalent to iter.filter_map(f).next().